R/join-by.R

```diff
@@ -0,0 +1,641 @@
+#' Join specifications
+#'
+#' @description
+#' `join_by()` constructs a specification to join by using a small domain
+#' specific language. The result can be supplied as the `by` argument to any
+#' of the join functions (such as [left_join()]).
+#'
+#' `join_by()` is constructed from a set of expressions, with each expression
+#' being separated by a comma. Each expression should reference a column from
+#' the left-hand table to match to the right-hand table, with some operator
+#' specifying how to match them together. Multiple expressions are combined
+#' using an "and" operation.
+#'
+#' ## Equi Joins:
```

Comment:
```suggestion
#' ## Equi joins:
```

We use sentence case for headings